Library/Homebrew/utils.sh

```diff
@@ -0,0 +1,47 @@
+puts_with_style() {
+  local message output_fd="$1" plain="$2" styled="$3"
+  shift 3
+
+  if [[ $# -eq 0 ]]
+  then
+    IFS= read -d '' -r message # Read message from `stdin`.
+    message="${message%$'\n'}" # Trim trailing newline.
+  else
+    message="$*"
+  fi
+
+  if [[ -t "$output_fd" ]] # If FD is a TTY, style message with color/bold/...
+  then
+    # shellcheck disable=SC2059
```

Comment:
[The warning being silenced here](https://github.com/koalaman/shellcheck/wiki/SC2059) is warranted when using user-provided or generated content as the format string for `printf`, but in our case the format string is fixed (coming from `ohai`, `oh1`, etc.), so the warning doesn't apply.